### PR TITLE
Fix `SubnetLaplace` backend check

### PR DESCRIPTION
**Pull request by wiseodd**
Based on #232

### CODE REVIEW

File: laplace/subnetlaplace.py [hunk 1]
```diff
@@ -99,11 +99,8 @@ def __init__(
             backend_kwargs=backend_kwargs,
         )
 
-        if backend is not None:
-            if not isinstance(backend, GGNInterface) and not isinstance(
-                backend, EFInterface
-            ):
-                raise ValueError("SubnetLaplace can only be used with GGN and EF.")
+        if backend is not None and not issubclass(backend, (GGNInterface, EFInterface)):
```
Comment (aleximmer) [inline laplace/subnetlaplace.py:102]: Use `isinstance` instead so the vanilla GGN/EF interfaces can be used or are they considered subclasses of themselves?

File: tests/test_subnetlaplace.py [hunk 2]
```diff
@@ -32,6 +36,17 @@
...
+    AsdlEF,
+    AsdlGGN,
+    AsdlHessian,
+    BackPackEF,
+    BackPackGGN,
+    AsdlHessian,
+    CurvlinopsEF,
+    CurvlinopsGGN,
+    CurvlinopsHessian,
+]
```
Comment (aleximmer) [inline tests/test_subnetlaplace.py:49]: Add vanilla backend?

**Review by aleximmer — APPROVED** [review]

File: laplace/subnetlaplace.py (hunk 1 above)
Comment (wiseodd) [reply laplace/subnetlaplace.py:102]: The latter is already true: https://docs.python.org/3/library/functions.html#issubclass

**Comment by aleximmer** [comment]
I think you can merge unless @runame wants to have a look as well.

**Comment by runame** [comment]
LGTM

**Review by runame — APPROVED** [review]
LGTM

**Pull request merged**